opencti-platform/opencti-graphql/src/database/file-storage.js

```diff
@@ -157,6 +158,46 @@ export const downloadFile = async (id) => {
   }
 };
 
+/**
+ * - Copy file from a place to another in S3
+ * - Store file in documents
+ * @param sourceId
+ * @param targetId
+ * @param sourceDocument
+ * @param targetEntityId
+ * @returns {Promise<null|void>} the document entity on success, null on errors.
+ */
+export const copyFile = async (sourceId, targetId, sourceDocument, targetEntityId) => {
+  try {
+    const input = {
+      Bucket: bucketName,
+      CopySource: `${bucketName}/${sourceId}`, // CopySource must start with bucket name, but not Key
+      Key: targetId
+    };
+    const command = new CopyObjectCommand(input);
+    await s3Client.send(command);
+    // Register in elastic
+    const targetMetadata = { ...sourceDocument.metaData, entity_id: targetEntityId };
+
+    const file = {
+      id: targetId,
+      name: sourceDocument.name,
+      size: sourceDocument.size,
+      information: '',
+      lastModified: new Date(),
+      lastModifiedSinceMin: sinceNowInMinutes(new Date()),
+      metaData: targetMetadata,
+      uploadStatus: 'complete',
+    };
+    await indexFileToDocument(file);
+    logApp.info(`[FILE STORAGE] Copy ${sourceId} to ${targetId} in success`, { document: file });
+    return file;
+  } catch (err) {
+    logApp.error(`[FILE STORAGE] Cannot copy file ${sourceId} to ${targetId} in S3`, { error: err });
```

Comment:
same, better to have a stable error message easy to search in logs and put all meta data like the ids in the object